src/Common/src/TypeSystem/Common/TypeDesc.cs

```diff
@@ -465,5 +481,25 @@ public virtual MethodDesc GetFinalizer()
         {
             return null;
         }
+
+        /// <summary>
+        /// Gets a value indicating whether this type has generic variance (the definition of the type
+        /// has a generic parameter that is co- or contravariant).
+        /// </summary>
+        public bool HasVariance
+        {
+            get
+            {
+                return (GetTypeFlags(TypeFlags.HasGenericVariance | TypeFlags.HasGenericVarianceComputed) & TypeFlags.HasGenericVariance) != 0;
+            }
+        }
+
+        /// <summary>
+        /// Gets the kind of this type.
+        /// </summary>
+        public abstract TypeKind Variety
```

Comment:
It seems to have a lot of overlap with TypeDesc.Category. Do we really need this? 

`IsDefType` can be `GetTypeFlags(TypeFlags.CategoryMask) <= TypeFlags.Interface` (also - it can assert that it is same as `type is DefType`).

`IsGenericParameter` can be `GetTypeFlags(TypeFlags.CategoryMask) == TypeFlags.GenericParameter`

The two places that call it directly can be switch over Category.